pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -194,12 +192,17 @@ func (r *InfraClusterController) ensureInfraCluster(ctx context.Context, log log
 			return nil, fmt.Errorf("error ensuring GCPCluster: %w", err)
 		}
 	case configv1.AzurePlatformType:
-		azureCluster := &azurev1.AzureCluster{}
-		if err := r.Get(ctx, client.ObjectKey{Namespace: defaultCAPINamespace, Name: r.Infra.Status.InfrastructureName}, azureCluster); err != nil && !kerrors.IsNotFound(err) {
-			return nil, fmt.Errorf("error getting InfraCluster object: %w", err)
+		if r.Infra.Status.PlatformStatus.Azure.CloudName == configv1.AzureStackCloud {
+			log.Info("%s cloud environment for platform %s is not supported", "environment", configv1.AzureStackCloud, "platform", configv1.AzurePlatformType)
+			return nil, nil //nolint:nilnil
```

Comment:
Should probably make this a sentinel error at some point, not blocking

Reply:
In which case, we should talk to the installer team to see if they know of any ill effect of implementing this change on a cluster from before that date. Perhaps backend pools were not in use before?

Reply:
Hey, after speaking with the installer team can confirm [Terraform was using the same value previously](https://github.com/openshift/installer/blob/release-4.16/data/data/azure/vnet/public-lb.tf#L101-L106), so this shouldn't break any clusters installed before it was introduced in the installer code :) 
Do you think we need to include this, though? I think we just want an object with the most minimum info possible to get up and running so it can be reconciled. Is it better to leave it out in that case if it's not necessary?

Reply:
As far as I know, this value is needed for control plane machines, so lets include it, if we know it